### ISSUE TITLE
Open the new-tab page with the container color and name

### ISSUE THREAD

**Issue by sagz** [chronakis/firefox-sticky-window-containers#1]
**Describe the bug**
Currently, after the first container tab, when the user opens a new tab (cmd+t or the new tab icon), the tab looks like it's from the default container.

**To Reproduce**
1. Open new window
2. Switch that first tab to some container
3. Close the older 'default' container tab
4. Create new tab with cmd+t / ctrl+t or new tab icon

**Expected behavior**
New tab should show the affordances to indicate that it's part of a container (tab color and name in URL-bar). It should mimic regular container behavior to retain user trust.

**Desktop**
 - OS: I tested on MacOS but I suspect this is universal
- Firefox 66.0.3
- Sticky Window Containers 1.0.1

**Comment by janmechtel** [comment]
I agree, actually I came to this issue list because I thought that the extension is not working as described.

I never tried to visit a site, i just expected the color in the tab color & name.

So for everyone else: after creating a new tab (ctrl+t) enter a url and go, only than will the color show up.

**Comment by chronakis** [comment]
Thank @sagz and @janmechtel you for reporting this.

The add-on was was based on the firefox-sticky-containers and the facebook container that determine the container when the tab navigates.

What you are saying makes sense and would be the nice thing to do. It needs a bit of investigation whether it is possible to assign the container to the tab when it is created.

cheers

Yiannis

**Comment by mindthefish** [comment]
I was short before uninstalling the add-on again before finding this issue. Thanks for clarifying, still would be nice to see the active container before navigating.

**Comment by RSully** [comment]
Any updates on this? I also thought it was broken

**Comment by chronakis** [comment]
No updates on this. As I mentioned above, it is a part of the way the tabs open and they container is determined "on navigation" rather than "on open tab".

It's a quite involved bit of work to change this behaviour and I am worried that it will take time to stabilise it and make reliable. It may sound funny, but I rely too much on this little add-on and my brain learned to ignore this.

I can't promise anything but I hope I will find the time and the motivation to develop this further soon.

**Comment by RSully** [comment]
Could you give any code/API pointers or how you might go about changing this? In case I or others want to try to change the behavior and submit a PR :)

**Comment by chronakis** [comment]
Hi @RSully ,

I will be happy to do this:

The source of the problem has been that once a tab is opened, you cannot update the cookie container that determines the tab's container. Also you I could not get an event of the type "before tab is created". So I hooked to the beforeNavigate event where there was enough information to determine if it's a facebook tab, a system tab or a plain tab. If it was plain, I would close the one with the default container and create a new tab of the desired container and then continue to navigation.

My first thought was to hook on the tab.onCreated event and then after it's created to change the cookie container. The problem was at this stage the URL of tab is not available and it lead to conflict with the facebook container and also the system tabs, e.g. preferences (endless loops).

I can see two directions:
1. Follow the above thought and solve the conflict with facebook and system tabs
2. Try to hook in the keyboard and menus to override the default new tab behaviour (probably a terrible idea)

There are a few comments in the code about this. I will try to look into more detail with some better understanding. It's been a while I looked at this.

**Comment by chronakis** [comment]
I have an update and if I am brave enough to risk pushing new bugs etc I will publish it. What this version has:

1. The container is decided when the tab opens, not when it navigates. There is the little flicker but the tab is having the correct colour.
2. There is now the option to use a menu to configure the window's default container (see image)
3. I added more robust customize and add-ons page detection.

<img width="438" alt="image" src="https://user-images.githubusercontent.com/493791/229365771-ac870eb1-cf6d-43c3-8a6c-0f503677e92d.png">

My biggest problem is getting this tested enough and avoid breaking it for everyone that uses it.
